### ISSUE TITLE
Problems with MySQL 5.5 in 1.1.1-pre

### ISSUE THREAD

**Issue by Nick-Lucas** [PomeloFoundation/Pomelo.EntityFrameworkCore.MySql#214]
Continuation of #210 

The current pre-release on Nuget is giving me "Database does not exist" on MySQL 5.5, even when it does exist and I'm trying to drop it.

This is when used with `.EnsureDeleted` and `.EnsureCreated`.

As suggested by @caleblloyd I have checked out the repo and run the [tests](https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/blob/master/test/Pomelo.EntityFrameworkCore.MySql.PerfTests/Commands/TestMigrateCommand.cs) written to capture this behaviour. When running `dotnet run testMigrate` I get the following. Is it possible there's an environment variable I need to set?
```
EnsureCreate creates database...Error in GeneratedTime.CreatedDateTime: DATETIME does not support values generated on Add or Update in MySql <= 5.5, try explicitly setting the column type to TIMESTAMP
   at Microsoft.EntityFrameworkCore.Migrations.MySqlMigrationsSqlGenerationHelper.ColumnDefinition(String schema, String table, String name, Type clrType, String type, Nullable`1 unicode, Nullable`1 maxLength, Boolean rowVersion, Boolean nullable, Object defaultValue, String defaultValueSql, String computedColumnSql, IAnnotatable annotatable, IModel model, MigrationCommandListBuilder builder) in C:\Users\nicho\Local\Repositories\Pomelo.EntityFrameworkCore.MySql\src\Pomelo.EntityFrameworkCore.MySql\Migrations\MySqlMigrationsSqlGenerator.cs:line 395
   at Microsoft.EntityFrameworkCore.Migrations.MigrationsSqlGenerator.ColumnDefinition(AddColumnOperation operation, IModel model, MigrationCommandListBuilder builder)
   at Microsoft.EntityFrameworkCore.Migrations.MigrationsSqlGenerator.Generate(CreateTableOperation operation, IModel model, MigrationCommandListBuilder builder, Boolean terminate)
   at Microsoft.EntityFrameworkCore.Migrations.MigrationsSqlGenerator.Generate(CreateTableOperation operation, IModel model, MigrationCommandListBuilder builder)
   at Microsoft.EntityFrameworkCore.Migrations.MigrationsSqlGenerator.Generate(MigrationOperation operation, IModel model, MigrationCommandListBuilder builder)
   at Microsoft.EntityFrameworkCore.Migrations.MySqlMigrationsSqlGenerationHelper.Generate(MigrationOperation operation, IModel model, MigrationCommandListBuilder builder) in C:\Users\nicho\Local\Repositories\Pomelo.EntityFrameworkCore.MySql\src\Pomelo.EntityFrameworkCore.MySql\Migrations\MySqlMigrationsSqlGenerator.cs:line 59
   at Microsoft.EntityFrameworkCore.Migrations.MigrationsSqlGenerator.Generate(IReadOnlyList`1 operations, IModel model)
   at Microsoft.EntityFrameworkCore.Storage.RelationalDatabaseCreator.CreateTables()
   at Microsoft.EntityFrameworkCore.Storage.RelationalDatabaseCreator.EnsureCreated()
   at Pomelo.EntityFrameworkCore.MySql.PerfTests.Commands.TestMigrateCommand.Run() in C:\Users\nicho\Local\Repositories\Pomelo.EntityFrameworkCore.MySql\test\Pomelo.EntityFrameworkCore.MySql.PerfTests\Commands\TestMigrateCommand.cs:line 15
   at Pomelo.EntityFrameworkCore.MySql.PerfTests.Commands.CommandRunner.Run(String[] args) in C:\Users\nicho\Local\Repositories\Pomelo.EntityFrameworkCore.MySql\test\Pomelo.EntityFrameworkCore.MySql.PerfTests\Commands\CommandRunner.cs:line 28
```

**Comment by caleblloyd** [comment]
I had to comment out all of the timestamp tests and JSON tests to get it working against 5.5.

Can you turn `Microsoft` logging to `Information` in your app and see what SQL is getting run?

**Comment by Nick-Lucas** [comment]
Oh so I'm getting my particular error because the database name is included in the connection string, which kind of makes sense. Not sure why it wasn't happening before, though.

If I remove it from the connection string, and use the fluent api's `modelBuilder.HasDefaultSchema('schema_name')`, then EnsureCreated gives me `MySql.Data.MySqlClient.MySqlException : No database selected`, which also sort of makes sense.

EnsureDeleted gives me this which, I assume from the stacktrace, is technically the same error.
```
System.ArgumentException : The string argument 'identifier' cannot be empty.

at Microsoft.EntityFrameworkCore.Utilities.Check.NotEmpty(String value, String parameterName)
   at Microsoft.EntityFrameworkCore.Storage.Internal.MySqlSqlGenerationHelper.DelimitIdentifier(String identifier)
   at Microsoft.EntityFrameworkCore.Migrations.MySqlMigrationsSqlGenerationHelper.Generate(MySqlDropDatabaseOperation operation, IModel model, MigrationCommandListBuilder builder)
   at Microsoft.EntityFrameworkCore.Migrations.MySqlMigrationsSqlGenerationHelper.Generate(MigrationOperation operation, IModel model, MigrationCommandListBuilder builder)
   at Microsoft.EntityFrameworkCore.Migrations.MigrationsSqlGenerator.Generate(IReadOnlyList`1 operations, IModel model)
   at Microsoft.EntityFrameworkCore.Storage.Internal.MySqlDatabaseCreator.Delete()
   at Microsoft.EntityFrameworkCore.Storage.RelationalDatabaseCreator.EnsureDeleted()
   at Tests.Integration.Pipelines.DbContexts.SalesDbContextFactory.Create(String dbmsName, DbContextFactoryOptions factoryOptions) in C:\Users\nicho\Local\Repositories\Pipelines\test\Tests.Integration.Pipelines\DbContexts\SalesDbContextFactory.cs:line 27
   at Tests.Integration.Pipelines.Providers.Sources.MySqlSourceTests..ctor() in C:\Users\nicho\Local\Repositories\Pipelines\test\Tests.Integration.Pipelines\Providers\Sources\MySqlSourceTests.cs:line 17
```

**Comment by Nick-Lucas** [comment]
If I run .EnsureCreated the following SQL also gets generated in my MySQL Query Log:
```
SELECT CASE WHEN COUNT(*) = 0 THEN FALSE ELSE TRUE END
                    FROM information_schema.tables
                    WHERE table_type = 'BASE TABLE' AND table_schema = ''
```
3 connections/change users
```
rollback
```
```
DROP PROCEDURE IF EXISTS mysql_ef_ensure_schema;
CREATE PROCEDURE mysql_ef_ensure_schema()
BEGIN
    IF NOT EXISTS (SELECT 1 FROM `INFORMATION_SCHEMA`.`SCHEMATA` WHERE `SCHEMA_NAME` = 'boop')
    THEN
        CREATE SCHEMA `boop`;
    END IF;
END;

CREATE TABLE `boop`.`Accounts` (
    `Id` int NOT NULL AUTO_INCREMENT,
    `Created` datetime NOT NULL,
    `FirstName` longtext,
    `LastName` longtext,
    CONSTRAINT `PK_Accounts` PRIMARY KEY (`Id`)
)
```
```
set session transaction isolation level repeatable read; start transaction
```

**Comment by Nick-Lucas** [comment]
So from this it looks like that first query is not being generated correctly after removing the database from the connection string, and with it in the connection string either MySQL or the provider is throwing

**Comment by caleblloyd** [comment]
Schema support through Fluent API has not been implemented yet, see #191.  It is planned for 2.0.0, there's a lot of testing that will need to go into it's implementation.

Ensure that you have a database in your connection string for the time being.

What SQL is logged when you do have a database in your connection string?

**Comment by Nick-Lucas** [comment]
Okay that makes sense then!

So with the connection string in place, and no database existing, all I get is this 'Connect' log:
```
dev@192.168.33.1 on my_schema
```

If I create the database manually then it does drop and recreate properly with .EnsureDeleted->.EnsureCreated

However, this is not the behaviour I was able to get after updating to MySQL-5.6 and Pomelo-1.1, or would expect a provider to give me.

So I'm thinking either MySQL 5.5 behaves differently than 5.6, and simply blocks connection, or the provider is throwing un-necessarily when the database doesn't exist.
